### ISSUE TITLE
Can't apply clientSitefilters

### ISSUE THREAD

**Issue by ugened47** [fluttercommunity/firestore_helpers#11]
Trying to query nearby drivers. Everything works pretty good until I'm trying to filter results with `clientSiteFilters`.

The code I'm using:
```dart
  Stream<List<dynamic>> nearbyDriversStream(area) {
    return getDataInArea(
      area: Area(area, 0.7),
      source: Firestore.instance.collection('drivers'),
      locationFieldNameInDB: 'position',
      mapper: (document) => document.data,
      locationAccessor: (item) => item['position'],
      distanceMapper: (item, dist) {
        item['distance'] = dist;
        return item;
      },
      clientSitefilters: [(item) => item['distance'] != 0.0], // Try to filter out current driver location
    );
  }

  _handleButtonPress() {
    nearbyDriversSubscription =
        nearbyDriversStream(currentLocation).listen((data) {
      print(data);
      nearbyDriversSubscription.cancel();
    });
  }
```
In this case I'm having an exception that says:
```
I/flutter (10905): ══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
I/flutter (10905): The following assertion was thrown while handling a gesture:
I/flutter (10905): type '() => List<(dynamic) => bool>' is not a subtype of type 'List<(dynamic) => bool>'
```
Compiler also complains if I'm trying to pass just `(item) => item['distance'] != 0.0`

What is the right way to use `clientSiteFilters`?

**Comment by escamoteur** [comment]
That should work as you do it although I don't recommend using it without proper static typing.
Also a comparison with 0.0 will probably fail because of precision errors. 
I have currently no idea, sorry.
Could you enable break on any exception and see where exactly the exception is thrown?

**Comment by ugened47** [comment]
@escamoteur Thank you for response!
I've also tried using typing that comes from this package like so:
```dart
  Stream<List<Map<String, dynamic>>> nearbyDriversStream(area) {
    List<ItemFilter<Map<String, dynamic>>> clientSitefilters = [
      (item) => item != null
    ]; // <--- List of filters
    return getDataInArea<Map<String, dynamic>>(
      area: Area(area, 0.7),
      source: Firestore.instance.collection('drivers'),
      locationFieldNameInDB: 'position',
      mapper: (document) => document.data,
      locationAccessor: (item) => item['position'],
      distanceMapper: (item, dist) {
        item['distance'] = dist;
        return item;
      },
      clientSitefilters:
          clientSitefilters, // Filter null values
    );
  }
```

The output is pretty the same. When use clientSitefilters the app throws:
```
flutter: ══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
flutter: The following assertion was thrown while handling a gesture:
flutter: type '() => List<(Map<String, dynamic>) => bool>' is not a subtype of type 'List<(Map<String,
flutter: dynamic>) => bool>'
flutter:
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new?template=BUG.md
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #0      getDataInArea (package:firestore_helpers/src/geo_helpers.dart:283:11)
flutter: #1      _HomeScreenState.nearbyDriversStream (package:taxoon_driver/routes/home.dart:117:12)
flutter: #2      _HomeScreenState._handleButtonPress (package:taxoon_driver/routes/home.dart:134:9)
flutter: #3      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:507:14)
```

Looks like error comes from `getDataInArea (package:firestore_helpers/src/geo_helpers.dart:283:11)`
It's `? () => clientSitefilters..insert(0, (item) => item != null)`
But I really don't know how to make it work as I'm not dart pro yet.

UPD: Just changed `? () => clientSitefilters..insert(0, (item) => item != null)` to `? clientSitefilters` and it worked. Now wondering how to properly make an insertion of default filter like it was before.

**Comment by escamoteur** [comment]
Should be fixed in V3.0.0
